### PR TITLE
[GVNSink] Avoid repeated hash lookups (NFC)

### PR DESCRIPTION
**Pull request by kazutakahirata**

### CODE REVIEW

File: llvm/lib/Transforms/Scalar/GVNSink.cpp [hunk 1]
```diff
@@ -535,12 +535,12 @@ class ValueTable {
...
       hash_code H = exp->getHashValue([=](Value *V) { return lookupOrAdd(V); });
-      auto I = HashNumbering.find(H);
-      if (I != HashNumbering.end()) {
+      auto [I, Inserted] = HashNumbering.try_emplace(H);
+      if (!Inserted) {
         e = I->second;
       } else {
         e = nextValueNumber++;
-        HashNumbering[H] = e;
+        I->second = e;
```
Comment (nikic) [inline llvm/lib/Transforms/Scalar/GVNSink.cpp:543]: Pass nextValueNumber to try_emplace instead?

Comment (kazutakahirata) [reply llvm/lib/Transforms/Scalar/GVNSink.cpp]: Addressed in the latest iteration.

**Comment by kazutakahirata** [comment]
Please take a look.  Thanks!

File: llvm/lib/Transforms/Scalar/GVNSink.cpp [hunk 2]
```diff
@@ -535,14 +535,10 @@ class ValueTable {
...
       hash_code H = exp->getHashValue([=](Value *V) { return lookupOrAdd(V); });
-      auto I = HashNumbering.find(H);
-      if (I != HashNumbering.end()) {
-        e = I->second;
-      } else {
-        e = nextValueNumber++;
-        HashNumbering[H] = e;
-        ExpressionNumbering[exp] = e;
-      }
+      auto [I, Inserted] = HashNumbering.try_emplace(H, nextValueNumber + 1);
```
Comment (nikic) [inline llvm/lib/Transforms/Scalar/GVNSink.cpp:538]: Shouldn't this be?
```suggestion
      auto [I, Inserted] = HashNumbering.try_emplace(H, nextValueNumber);
```
The old code is using the result of post-inc, which is the pre-increment value.

Comment (kazutakahirata) [reply llvm/lib/Transforms/Scalar/GVNSink.cpp]: Oops.  Thanks for catching this!  Fixed in the latest iteration.

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**